src/Package/Impl/DataInspect/VisualGrid/MatrixViewAutomationPeer.cs

```diff
@@ -0,0 +1,145 @@
+                    return this;
+                case PatternInterface.Scroll:
+                    return ScrollProvider;
+                default:
+                    return base.GetPattern(patternInterface);
+            }
+        }
+
+        public IRawElementProviderSimple GetItem(int row, int column)
+            => ProviderFromPeer(GetOrCreateCell(row, column));
+
+        public void Update() {
+            var scroller = _owner.Scroller;
+            var oldVisibleRange = _visibleRange;
+            _visibleRange = scroller?.DataViewport ?? new GridRange(new Range(0, 0), new Range(0, 0));
+
+            if (!oldVisibleRange.Contains(_visibleRange)) {
+                ResetChildrenCache();
+            }
+            
+            foreach (var automationPeer in GetChildren()) {
+                switch (automationPeer) {
+                    case MatrixViewCellAutomationPeer cellAutomationPeer when _visibleRange.Contains(cellAutomationPeer.Row, cellAutomationPeer.Column):
+                        cellAutomationPeer.SetValue(scroller?.CellsData[cellAutomationPeer.Row, cellAutomationPeer.Column]);
+                        break;
+                    case MatrixViewHeaderAutomationPeer headerAutomationPeer when _visibleRange.Contains(_visibleRange.Rows.Start, headerAutomationPeer.Column):
+                        headerAutomationPeer.SetValue(scroller?.ColumnsData[0, headerAutomationPeer.Column]);
+                        break;
+                    case MatrixViewItemAutomationPeer itemAutomationPeer:
+                        itemAutomationPeer.ClearValue();
+                        break;
+                }
+            }
+        }
+
+        public bool IsInVisibleRange(int row, int column) => _visibleRange.Contains(row, column);
+
+        protected override List<AutomationPeer> GetChildrenCore() {
+            var children = new List<AutomationPeer>();
+
+            for (var row = _visibleRange.Rows.Start.ReduceToInt(); row < _visibleRange.Rows.End.ReduceToInt(); row++) {
+                for (var column = _visibleRange.Columns.Start.ReduceToInt(); column < _visibleRange.Columns.End.ReduceToInt(); column++) {
+                    children.Add(GetOrCreateCell(row, column));
+                }
+            }
+
+            for (var column = _visibleRange.Columns.Start.ReduceToInt(); column < _visibleRange.Columns.End.ReduceToInt(); column++) {
+                children.Add(GetOrCreateHeader(column));
+            }
+
+            PurgePeersCollections();
+
+            return children;
+        }
+        
+        public IRawElementProviderSimple[] GetRowHeaders() => null;
+
+        public IRawElementProviderSimple[] GetColumnHeaders() {
+            var start = _visibleRange.Columns.Start.ReduceToInt();
+            var count = _visibleRange.Columns.End.ReduceToInt() - start;
+            var headers = new IRawElementProviderSimple[count];
+            for (var i = 0; i < headers.Length; i++) {
```

Comment:
In LINQ, it would be something like:
```cs
_visibleRange.Columns.GetEnumerable()
    .Where(i => i <= int.MaxValue)
    .Cast<int>()
    .Select(i => ProviderFromPeer(GetOrCreateHeader(i)))
    .ToArray();
```

Can't say I like it more than existing code